quinn-udp/tests/tests.rs

```diff
@@ -240,7 +243,23 @@ fn test_send_recv(send: &Socket, recv: &Socket, transmit: Transmit) {
                 ),
             }
         }
-        assert_eq!(meta.ecn, transmit.ecn);
+
+        if match transmit.destination.ip() {
+            IpAddr::V4(_) => true,
+            IpAddr::V6(a) => a.to_ipv4_mapped().is_some(),
+        } && cfg!(target_os = "android")
+            && std::env::var("API_LEVEL")
+                .ok()
+                .and_then(|v| v.parse::<u32>().ok())
+                .expect("API_LEVEL environment variable to be set on Android")
+                <= 25
+        {
```

Comment:
Nit: don't love this multi-line, multi-level-of-indent `if` condition. Maybe bind the first clause and the `api_level` value?

I find the `expect()` explanation a little confusing since `[cfg_attr(
    not(any(target_os = "linux", target_os = "windows", target_os = "android")),
    ignore
)]` does not seem to restrict this test to Android only?

Reply:
> Maybe bind the first clause

Done.

> and the `api_level` value?

Can't do that, as `API_LEVEL` will only be available on Android.



> I find the `expect()` explanation a little confusing since `[cfg_attr( not(any(target_os = "linux", target_os = "windows", target_os = "android")), ignore )]` does not seem to restrict this test to Android only?

Given that Rust short-circuits boolean operations (aka. [lazy boolean operators](https://doc.rust-lang.org/reference/expressions/operator-expr.html#lazy-boolean-operators)) the third boolean expression is only evaluated if the first and second is `true`.